### ISSUE TITLE
Standardize on Web APIs (`Request` / `Response` / `URL` / etc)

### ISSUE THREAD

**Issue by thescientist13** [ProjectEvergreen/greenwood#948]
### Discussed in https://github.com/ProjectEvergreen/greenwood/discussions/867

<div type='discussions-op-text'>

<sup>Originally posted by **thescientist13** January 13, 2022</sup>
I would have to do a little more research into Koa, but the [**Fastify**](https://www.fastify.io/) project is looking super interesting and is actively lead / maintained by one the TSC members of NodeJS.  As Greenwood moves towards #708 / #576 , it would be good to evaluate the server solution we are using, as it was originally only considered under the basis of being a _dev_ server, is now going to be put into production situations, and so performance and speed will play a more important role in user experience.

In addition, we should really standardize our [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request) / [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) objects.</div>

Should also evaluate native [`fetch` from Node](https://blog.logrocket.com/fetch-api-node-js/) and also keep an eye on [WINTER CG](https://wintercg.org/).

----

So this change would definitely impact plugin resources the most (not necessarily in a bad way) and coming out of #974 , it became clear that `ResourceInterface` may need to be deprecated:
- The assumption of having default implementations on `ResourceInterface` like `shouldResolve` while convenient, has cause [hard to trace issues where plugins were running in lifecycles automatically](https://github.com/ProjectEvergreen/greenwood/pull/974/files#diff-e1c21a418290cc7cfabe43827e44d70e31dc684116aa943bdbd49410e8e1abb3R16) and thus manipulating the req / res unknowingly.  You can see in this PR where i had to explicitly opt-out.  In actuality, it was more of an Abstract Class rather than an interface, so it was even badly named.  I think I would prefer to just get rid of it, but I think `contentType` and `extension` could still be useful maybe?
- With #930 , this plugin will extend past just the browser.  So resource plugins are just generic handlers at that point, environment agnostic.
- Should better explain the nuance of changing the `Request` vs the `Response` and what the order of operations are done by Greenwood when taking the response from resource plugin and setting up the next resource plugin so it can operate on it.  Should it be assumed that `Request` params are immutable, and `Response` objects are mutable?

----

Basically, we should be doing more of:
1. [ ] Leverage `URL` top to bottom internally (instead of `path`)
1. [ ] Use `Request` / `Response` everywhere, at very minimum for plugins / resources
1. [ ] Find a Web standard focused web server (this should be Fastify)

Should also hopefully address #949
